akka-http-core/src/main/scala/akka/http/impl/engine/http2/framing/RSTFrameLimit.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2.framing
+
+import akka.annotation.InternalApi
+import akka.http.impl.engine.http2.{ FrameEvent, Http2Compliance }
+import akka.http.impl.engine.http2.FrameEvent.RstStreamFrame
+import akka.http.impl.engine.http2.Http2Protocol.ErrorCode
+import akka.http.scaladsl.settings.Http2ServerSettings
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+
+import scala.concurrent.duration._
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final class RSTFrameLimit(http2ServerSettings: Http2ServerSettings) extends GraphStage[FlowShape[FrameEvent, FrameEvent]] {
+
+  private val maxResets = http2ServerSettings.maxResets
+  private val maxResetsIntervalNanos = http2ServerSettings.maxResetsInterval.toNanos
+
+  val in = Inlet[FrameEvent]("in")
+  val out = Outlet[FrameEvent]("out")
+  val shape = FlowShape(in, out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with InHandler with OutHandler {
+    private var rstCount = 0
+    private var rstSpanStartNanos = 0L
+
+    setHandlers(in, out, this)
+
+    override def onPush(): Unit = {
+      grab(in) match {
+        case frame: RstStreamFrame =>
+          rstCount += 1
+          val now = System.nanoTime()
+          if (rstSpanStartNanos == 0L) {
```

Comment:
0 is also a valid value for `System.nanoTime`. Doesn't really matter I guess.
We don't want to `rstSpanStartNanos = System.nanoTime` in the constructor, because there might not be any Rst?

Reply:
That could be fine I guess, it is only once per connection in that case, so other things will be way more expensive anyway.

Reply:
Added a boolean